packages/oasis3-mct/package.py

```diff
@@ -180,18 +197,140 @@ def __create_pkgconfig(self, spec, prefix):
             if nchars_written < len(text):
                 raise OSError
 
+    # https://spack-tutorial.readthedocs.io/en/ecp21/tutorial_advanced_packaging.html
+    @property
+    def libs(self):
+        # NOTE: The order matters during the linking step of the cice5 build!
+        # "-lpsmile.MPI1 -lmct -lmpeu -lscrip"
+        libraries = ["libpsmile.MPI1", "libmct", "libmpeu", "libscrip"]
+        return find_libraries(
+            libraries, root=self.prefix, shared=False, recursive=True
+        )
+
+    # Following pattern:
+    # https://spack-tutorial.readthedocs.io/en/latest/tutorial_buildsystems.html
+    def edit(self, spec, prefix):
+        SRCDIR = self.stage.source_path
+        makeinc_path = join_path(SRCDIR, self.__makeinc)
+        config = {}
+
+        # TODO: Clean up the debug / non-debug mechanism once the compiler
+        #       flag audit is finished.
+        #       https://github.com/ACCESS-NRI/ACCESS-OM/issues/12
+        NCI_INTEL_FLAGS = "-r8 -i4 -traceback -fpe0 -convert big_endian -fno-alias -ip -check noarg_temp_created"
+        NCI_OPTIM_FLAGS = "-g3 -O2 -axCORE-AVX2 -debug all -check none -qopt-report=5 -qopt-report-annotate"
+        CFLAGS = ""
+        if "+nodebuginfo" in self.spec:
+            NCI_INTEL_FLAGS = "-r8 -i4 -fpe0 -convert big_endian -fno-alias -ip -check noarg_temp_created"
+            NCI_OPTIM_FLAGS = "-g0 -O0 -axCORE-AVX2 -debug none -check none -qopt-report=5 -qopt-report-annotate"
+            CFLAGS = "-g0"
```

Comment:
```suggestion
        if "+nodebuginfo" in self.spec:
            NCI_OPTIM_FLAGS = " ".join(NCI_OPTIM_FLAGS, "-g0 -debug none")
            CFLAGS = "-g0"
        else:
            NCI_INTEL_FLAGS = " ".join(NCI_INTEL_FLAGS, "-traceback")            
            NCI_OPTIM_FLAGS = " ".join(NCI_OPTIM_FLAGS, "-g3 -debug all")
```
I'm suggesting setting the common flags in one location and just adding the debug option here. Maybe this is what your TODO above is referencing? 

Could you check I've got the common flags correct? I assumed the `-O0` at line 225 was a mistake?

Reply:
Please comment in https://github.com/ACCESS-NRI/ACCESS-OM/issues/12

Reply:
Have done so but this

> Could you check I've got the common flags correct? I assumed the -O0 at line 225 was a mistake?

Is just a mistake I think. From https://github.com/ACCESS-NRI/ACCESS-OM/issues/12#issuecomment-1428923689 the `make.nci` has this:
```
ifeq ($(DEBUG), yes)
    NCI_DEBUG_FLAGS = -g3 -O0 -fpe0 -no-vec -debug all -check all -no-vec
    F90FLAGS_1      = $(NCI_INTEL_FLAGS) $(NCI_REPRO_FLAGS) $(NCI_DEBUG_FLAGS)
    CPPDEF          = -Duse_netCDF -Duse_comm_$(CHAN) -DTREAT_OVERLAY -DDEBUG -D__VERBOSE
    MCT_FCFLAGS     = $(NCI_REPRO_FLAGS) $(NCI_DEBUG_FLAGS) -ip
else
    NCI_OPTIM_FLAGS = -g3 -O2 -axCORE-AVX2 -debug all -check none -qopt-report=5 -qopt-report-annotate
    F90FLAGS_1      = $(NCI_INTEL_FLAGS) $(NCI_REPRO_FLAGS) $(NCI_OPTIM_FLAGS)
    CPPDEF          = -Duse_netCDF -Duse_comm_$(CHAN) -DTREAT_OVERLAY
    MCT_FCFLAGS     = $(NCI_REPRO_FLAGS) $(NCI_OPTIM_FLAGS) -ip
endif
```

Specifically:
```make
NCI_OPTIM_FLAGS = -g3 -O2 -axCORE-AVX2 -debug all -check none -qopt-report=5 -qopt-report-annotate
```
versus
```python
NCI_OPTIM_FLAGS = "-g0 -O0 -axCORE-AVX2 -debug none -check none -qopt-report=5 -qopt-report-annotate"
```

I understand some of those flags are changed for build repro reasons, but surely it should be `-O2` in this case?

Reply:
# TODO: Clean up the debug / non-debug mechanism once the compiler
        #       flag audit is finished.
        #       https://github.com/ACCESS-NRI/ACCESS-OM/issues/12